libgalois/include/katana/PropertyGraph.h

```diff
@@ -299,35 +346,109 @@ class KATANA_EXPORT PropertyGraph {
   std::string ReportDiff(const PropertyGraph* other) const;
 
   std::shared_ptr<arrow::Schema> node_schema() const {
-    return rdg_.node_properties()->schema();
+    return node_properties()->schema();
   }
 
   std::shared_ptr<arrow::Schema> edge_schema() const {
-    return rdg_.edge_properties()->schema();
+    return edge_properties()->schema();
   }
 
+  /// \returns the number of node types
+  size_t GetNodeTypesNum() const { return node_type_to_type_ids_.size(); }
```

Comment:
I agree but I wanted to keep the names in the class consistent. This class uses `Get` in the function names and uses `Num` in the end, so I stuck to that convention. We can make a separate pass/PR to change the function names of this class.

Reply:
ok. Do you want to capture this as an issue and resolve all such comments here?

Reply:
The relevant bit is: https://google.github.io/styleguide/cppguide.html#Function_Names and https://google.github.io/styleguide/cppguide.html#Exceptions_to_Naming_Rules

If a function behaves like a field access (e.g., constant time, no exceptions), it may be named like a variable `num_node_types()`. If a method is intended to mirror some other existing interface, it may also be in snake case (e.g., `.size()`, `.find_if()`).

As to `GetFoo()` vs `Foo()`, I would use `GetFoo` although there is no specific requirement in the Google style guide to do this AFAIK. To me it seems implied because if you didn't like `GetFoo` you would just name the function like a variable `foo()` and in general, people name functions as active verbs.

Let's not churn the codebase unnecessarily. The reason we have a style guide is so that people can spend their time discussing important functional requirements of code rather than arbitrary style nits.

Reply:
This goes a bit beyond just stylistic issues. What I want to discourage is setters. One of the gifts of Java is mindless generation of getters and setters (IDEs automate it for you). Ideally, a setter should be a moment to pause and think: do we really need this? If the answer is yes, there's usually a better name for the setter that conveys the intent better. So my line of defense is to avoid getters so that you don't ever get to setters, but I'm OK with keeping getters (with name GetBlah) in the interest of avoiding purely stylistic changes.

Reply:
> What I want to discourage is setters.

That is fine but there is a limit to what can and should be done in the context of a particular PR, and as far as I can see, most comments have been about getters so far.

If "no setters" is important to you, there probably should be wider discussion so that we can reach some agreement and enforce it throughout the codebase; otherwise nothing will converge.

Reply:
Yes, the discussion is beyond the scope of this PR and we should have that discussion. I've just started looking at the code and within PropertyGraph.h, I see a bunch of setters and two conventions for getters (getters with and without get). Not sure why we have both the styles? Is it because one API is for the partitioner or underlying substack while the other is for the stack above? 
I'm for consistent and clean interfaces. We can have a discussion about cost and benefits of such a change, but to me a proliferation of getters and setters is a sign of degrading quality (degrading towards typical Java code :)).

Reply:
Is the method naming in PropertyGraph inconsistent (some are lower case and some start with uppercase) or am I missing some convention being followed here?

Reply:
Didn't Donald answer that question in this thread above? If he didn't, can you please be more specific in your question?

Reply:
Well, the discussion with @ddn0 was regarding (elimination of) setters, which is a separate issue that's not high priority but important to me. That aside, there's still the issue of two styles of method names. Is there a reason or is it something to be fixed? If it's something we should fix then we can create an issue.

Reply:
Let me summarize.

There are two ways to name a method.
- PascalCase() - most common, active verb phrase
- snake_case() - rarer, noun-phrase, should behave like a field access: O(1), no-exceptions, etc.

Reply:
I see. More reasons to hate GSG, I guess.